<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Iteration Analysis of Sparse Setting with Low LD among causal variants |r| &lt; 0.25</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SuSiE-ASH</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="improvements.html">SuSiE Improvements</a>
</li>
<li>
  <a href="vignette_ash_failure.html">Vignette: mr.ash Failure</a>
</li>
<li>
  <a href="vignette_iteration_analysis_1.html">Vignette: Iteration Analysis</a>
</li>
<li>
  <a href="synthetic_association.html">Synthetic Association</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/alexmccreight/SuSiE-ASH">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Iteration Analysis of Sparse Setting with
Low LD among causal variants |r| &lt; 0.25</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-03-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>SuSiE-ASH/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240328code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240328)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240328code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240328)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/alexmccreight/Columbia/Research/SuSiE-ASH/SuSiE-ASH/code/susie_versions/SuSiE_Ash_Marginal.R
</td>
<td style="text-align:left;">
code/susie_versions/SuSiE_Ash_Marginal.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/alexmccreight/Columbia/Research/SuSiE-ASH/SuSiE-ASH/new-rcpp/mr.ash.alpha
</td>
<td style="text-align:left;">
new-rcpp/mr.ash.alpha
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/alexmccreight/Columbia/Research/SuSiE-ASH/SuSiE-ASH/submodules/susieR
</td>
<td style="text-align:left;">
submodules/susieR
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/alexmccreight/Columbia/Research/SuSiE-ASH/SuSiE-ASH/code/susie_versions/susie_inf.R
</td>
<td style="text-align:left;">
code/susie_versions/susie_inf.R
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/alexmccreight/Columbia/Research/SuSiE-ASH/SuSiE-ASH/X_subset
</td>
<td style="text-align:left;">
X_subset
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomalexmccreightSuSiEASHtree1c6083ec1fa5d14fe0513c3ac9ba9338a9aea6betargetblank1c6083ea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/alexmccreight/SuSiE-ASH/tree/1c6083ec1fa5d14fe0513c3ac9ba9338a9aea6be" target="_blank">1c6083e</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomalexmccreightSuSiEASHtree1c6083ec1fa5d14fe0513c3ac9ba9338a9aea6betargetblank1c6083ea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/alexmccreight/SuSiE-ASH/tree/1c6083ec1fa5d14fe0513c3ac9ba9338a9aea6be" target="_blank">1c6083e</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    code/.DS_Store
    Ignored:    code/susie_versions/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    manuscript_scripts/
    Ignored:    new-rcpp/.DS_Store
    Ignored:    new-rcpp/matrix.multiplication/.DS_Store
    Ignored:    new-rcpp/matrix.multiplication/.Rapp.history
    Ignored:    new-rcpp/mr.ash.alpha/.DS_Store
    Ignored:    new-rcpp/mr.ash.alpha/.Rbuildignore
    Ignored:    new-rcpp/mr.ash.alpha/R/.DS_Store
    Ignored:    new-rcpp/mr.ash.alpha/src/.DS_Store
    Ignored:    new-rcpp/mr.ash.alpha/src/RcppExports.o
    Ignored:    new-rcpp/mr.ash.alpha/src/caisa_rcpp.o
    Ignored:    new-rcpp/mr.ash.alpha/src/mr.ash.alpha.so
    Ignored:    scripts/.DS_Store
    Ignored:    scripts/Old Scripts/.DS_Store
    Ignored:    scripts/benchmark/.DS_Store
    Ignored:    simulation/.DS_Store
    Ignored:    simulation/misc/.DS_Store
    Ignored:    submodules/.DS_Store

Untracked files:
    Untracked:  X_subset
    Untracked:  code/Tmp.Rmd
    Untracked:  misc/
    Untracked:  oligogenic_data_generation.R
    Untracked:  overleaf_graphics/
    Untracked:  preprint/KISS-paper.Rmd
    Untracked:  scripts/LD_blocks/
    Untracked:  scripts/benchmark/command_generation/
    Untracked:  scripts/benchmark/plots/
    Untracked:  scripts/benchmark/sparse_metrics/
    Untracked:  scripts/benchmark/sparse_result_files/
    Untracked:  scripts/benchmark/sparse_result_files_2/
    Untracked:  scripts/commands_to_submit_eqtl_simple.R
    Untracked:  scripts/eqtl_simulation_marginal.R
    Untracked:  scripts/eqtl_simulation_simple.R
    Untracked:  scripts/eqtl_simulation_v2.R
    Untracked:  scripts/fineboost/
    Untracked:  scripts/sparse_blip.R
    Untracked:  scripts/sparse_eqtl_simulation_marginal.R
    Untracked:  scripts/sparse_eqtl_simulation_simple.R
    Untracked:  scripts/susie_ash_mod/
    Untracked:  seed9_data.rds
    Untracked:  seed9_ld.rds
    Untracked:  seed9_res.rds
    Untracked:  simulation/15x3jxl3enlnnsejfeke1_tmate.log
    Untracked:  simulation/15x3jxl3enlnnsejfeke1_tmate_session.log
    Untracked:  simulation/eqtl-results/
    Untracked:  simulation/misc/Rcpp-Comparison.R
    Untracked:  simulation/misc/RcppTesting.Rmd
    Untracked:  simulation/misc/inf_debugging.Rmd
    Untracked:  simulation/misc/learner.Rmd
    Untracked:  simulation/misc/sim_analysis.Rmd
    Untracked:  simulation/misc/sim_debugging.Rmd
    Untracked:  simulation/misc/sim_download.Rmd
    Untracked:  simulation/sparse-results/
    Untracked:  sparse_simulation.ipynb
    Untracked:  test_script.R
    Untracked:  tmp_files/
    Untracked:  vignettes/

Unstaged changes:
    Modified:   code/susie_versions/SuSiE_Ash_Marginal.R
    Modified:   code/susie_versions/susie_ash_mod.R
    Modified:   code/susie_versions/susie_inf.R
    Deleted:    scripts/LD_blocks_precomputations.R
    Deleted:    scripts/comannds_to_submit_LD.R
    Deleted:    scripts/commands_to_submit_LD_precomputations.R
    Deleted:    scripts/commands_to_submit_ash_mod.R
    Modified:   scripts/commands_to_submit_eqtl.R
    Deleted:    scripts/eqtl_ash_mod.R
    Deleted:    scripts/eqtl_ash_mod_default_grid.R
    Modified:   scripts/eqtl_simulation.R
    Deleted:    scripts/process_ld_block.R
    Modified:   scripts/sparse/commands_to_submit_sparse.R
    Modified:   simulation/LD_blocks.sh
    Modified:   simulation/commands_to_submit.txt
    Modified:   simulation/eqtl_sim.sh
    Modified:   simulation/identification.Rmd
    Modified:   simulation/sparse_eqtl_sim.sh
    Modified:   submodules/VEB.Boost
    Modified:   submodules/pecotmr

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/vignette_iteration_analysis_1.Rmd</code>) and HTML
(<code>docs/vignette_iteration_analysis_1.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexmccreight/SuSiE-ASH/42bff018856a73a2c8529044430d71018dad43ae/docs/vignette_iteration_analysis_1.html" target="_blank">42bff01</a>
</td>
<td>
alexmccreight
</td>
<td>
2025-02-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexmccreight/SuSiE-ASH/608213655f6e9391da87a2a611fd86216230d766/docs/vignette_iteration_analysis_1.html" target="_blank">6082136</a>
</td>
<td>
alexmccreight
</td>
<td>
2025-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/alexmccreight/SuSiE-ASH/acb1163e2abae63ee7289861ab6d9e9854f592af/docs/vignette_iteration_analysis_1.html" target="_blank">acb1163</a>
</td>
<td>
alexmccreight
</td>
<td>
2025-02-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/alexmccreight/SuSiE-ASH/blob/15fab0ed5066bc0b72b120b0a6f40ae1303f1853/analysis/vignette_iteration_analysis_1.Rmd" target="_blank">15fab0e</a>
</td>
<td>
alexmccreight
</td>
<td>
2025-02-26
</td>
<td>
workflowr::wflow_publish(files =
"analysis/vignette_iteration_analysis_1.rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>In this iteration analysis, we investigate how the posterior
inclusion probability (alpha) plots &amp; credible sets (CS) evolve
across successive iterations of SuSiE, SuSiE-inf, and SuSiE.ash. Our aim
is to determine whether the alpha values begin to vary early on or only
after several iterations.</p>
<p>For clarity, only effect groups (L) that have at least one alpha
value exceeding 0.05 are included in the legend, in order to avoid 20
groups cluttering the plot. Additionally, causal variants are marked
with dots along the bottom of the graph–green if the variant is captured
within a CS and red otherwise–with the dot sizes scaled in proportion to
the magnitude of the causal effect. Finally, for any non-causal effect
with an alpha value exceeding 0.25, we annotate its highest LD, measured
by absolute correlation, among any of the causal variants.</p>
<pre class="r"><code># Data &amp; Precomputations
X_subset &lt;- readRDS(&quot;/Users/alexmccreight/Columbia/Research/SuSiE-ASH/SuSiE-ASH/X_subset&quot;)
LD_mat &lt;- cor(X_subset)
colnames(LD_mat) &lt;- 1:ncol(LD_mat)
rownames(LD_mat) &lt;- 1:nrow(LD_mat)
scaled_X_full &lt;- scale(X_subset)
n_samples &lt;- nrow(scaled_X_full)
XtX &lt;- t(scaled_X_full) %*% scaled_X_full
LD  &lt;- XtX / n_samples
eig &lt;- eigen(LD, symmetric = TRUE)
V   &lt;- eig$vectors[, ncol(eig$vectors):1]
Dsq &lt;- pmax(n_samples * sort(eig$values), 0)
VtXt &lt;- t(V) %*% t(scaled_X_full)

# Helper Functions
plot_susie_fit &lt;- function(fit, data, model = c(&quot;original&quot;, &quot;inf&quot;, &quot;ash&quot;), 
                             threshold = 0.05, set_par = TRUE,
                             global_alpha_max = NULL) {
  model &lt;- match.arg(model)
  
  # Extract the appropriate PIP matrix
  if (model == &quot;original&quot;) {
    mat &lt;- fit$alpha  # L x p
    title_model &lt;- &quot;SuSiE&quot;
  } else if (model %in% c(&quot;inf&quot;, &quot;ash&quot;)) {
    mat &lt;- t(fit$PIP) # p x L
    title_model &lt;- if (model == &quot;inf&quot;) &quot;SuSiE-inf&quot; else &quot;SuSiE-ash&quot;
  } else {
    stop(&quot;Invalid model type. Please choose &#39;original&#39;, &#39;inf&#39;, or &#39;ash&#39;.&quot;)
  }
  
  # Filter groups (rows) by threshold.
  max_alpha &lt;- apply(mat, 1, max)
  keep &lt;- max_alpha &gt;= threshold
  alpha_filtered &lt;- mat[keep, , drop = FALSE]
  
  # Determine the y-axis limits.
  # If a global maximum is provided, use that; otherwise use the maximum in this fit.
  if (is.null(global_alpha_max)) {
    global_alpha_max &lt;- max(alpha_filtered)
  }

    ylim_val &lt;- c(0, global_alpha_max * 1.1)
  
  # Color palette for groups.
  if (!requireNamespace(&quot;RColorBrewer&quot;, quietly = TRUE)) {
    stop(&quot;RColorBrewer package is required but not installed.&quot;)
  }
  library(RColorBrewer)
  num_groups &lt;- nrow(alpha_filtered)
  if (num_groups &lt;= 8) {
    base_group_colors &lt;- brewer.pal(num_groups, &quot;Set1&quot;)
  } else {
    base_group_colors &lt;- colorRampPalette(brewer.pal(9, &quot;Set1&quot;))(num_groups)
  }
  
  # Build a color matrix for the barplot.
  num_variants &lt;- ncol(alpha_filtered)
  col_matrix &lt;- matrix(rep(base_group_colors, each = num_variants), 
                       nrow = num_groups, ncol = num_variants)
  col_matrix &lt;- adjustcolor(col_matrix, alpha.f = 1)
  
  # Optionally set plotting parameters.
  if (set_par) {
    par(mar = c(5, 4, 4, 8), xpd = TRUE)
  }
  
  # Create the barplot
  bp &lt;- barplot(alpha_filtered,
                beside = TRUE,
                col = col_matrix,
                border = NA,
                density = NA,
                names.arg = 1:num_variants,
                xlab = &quot;Variable index (p)&quot;,
                ylab = &quot;Posterior Inclusion Probability (alpha)&quot;,
                main = paste(title_model, &quot;- Filtered Posterior Inclusion Probabilities by Group&quot;),
                ylim = ylim_val)
  
  # Compute the x positions (centers) for each cluster of bars.
  cluster_x &lt;- colMeans(bp)
  y_marker &lt;- -0.02
  
  # Compute dot sizes based on absolute effect sizes for causal variants.
  dot_cex &lt;- abs(data$beta[data$causal_indices])
  if (max(dot_cex) &gt; 0) {
    dot_cex &lt;- dot_cex / max(dot_cex) * 1.25
  } else {
    dot_cex &lt;- rep(1, length(dot_cex))
  }
  
  # Determine credible set variants.
  if (model == &quot;original&quot;) {
    cs_variants &lt;- unlist(fit$sets$cs)
  } else {
    cs_variants &lt;- unlist(fit$sets)
  }
  
  # Add causal index points: if the causal variant is in the credible set, plot in green; else red.
  causal_indices &lt;- data$causal_indices
  dot_colors &lt;- ifelse(causal_indices %in% cs_variants, &quot;green&quot;, &quot;red&quot;)
  points(cluster_x[causal_indices],
         rep(y_marker, length(causal_indices)),
         pch = 19,
         col = dot_colors,
         cex = dot_cex)
  
  # Add legend for groups.
  legend(&quot;topright&quot;,
         inset = c(-0.25, 0),
         legend = paste(&quot;Group&quot;, which(keep)),
         fill = base_group_colors[which(keep)],
         cex  = 0.8,
         ncol = 1)
  
  # --- False Positive LD Annotations ---
  # Identify false positives: variants with max PIP &gt;= 0.25 that are not causal.
  all_high_pip &lt;- which(apply(alpha_filtered, 2, max) &gt;= 0.25)
  false_positives &lt;- setdiff(all_high_pip, causal_indices)
  
  if (length(false_positives) &gt; 0) {
    sorted_fp &lt;- sort(false_positives)
    clusters &lt;- list()
    current_cluster &lt;- c(sorted_fp[1])
    if(length(sorted_fp) &gt; 1){
      for(i in 2:length(sorted_fp)){
        if(sorted_fp[i] - tail(current_cluster, 1) &lt;= 30){
          current_cluster &lt;- c(current_cluster, sorted_fp[i])
        } else {
          clusters[[length(clusters) + 1]] &lt;- current_cluster
          current_cluster &lt;- c(sorted_fp[i])
        }
      }
    }
    clusters[[length(clusters) + 1]] &lt;- current_cluster
    
    for(cluster in clusters) {
      cluster_x_val &lt;- mean(cluster_x[cluster])
      r_values &lt;- sapply(cluster, function(fp) {
        max(abs(LD_mat[fp, causal_indices]))
      })
      lbl &lt;- paste0(&quot;ID = {&quot;, paste(cluster, collapse = &quot;, &quot;), &quot;}, |r| = {&quot;, 
                    paste(round(r_values, 2), collapse = &quot;, &quot;), &quot;}&quot;)
      cluster_y &lt;- mean(sapply(cluster, function(fp) { max(alpha_filtered[, fp]) }))
      cluster_y_offset &lt;- cluster_y + 0.02
      text(x = cluster_x_val, y = cluster_y_offset, labels = lbl, pos = 3, 
           cex = 0.7, srt = 0, col = &quot;black&quot;)
    }
  }
  
  # --- Residual Variance Estimates ---
  #   susie_fit$sigma2 is on the original scale, while susie_inf and susie_ash (fit$sigmasq) are on the standardized scale.
  #   Similarly, data$var_epsilon is on the original scale.
  # We want all to be on the standardized  scale.
  sd_y &lt;- sd(data$y)
  if (model == &quot;original&quot;) {
    sigma2_std &lt;- fit$sigma2[1] / (sd_y^2)
  } else {
    sigma2_std &lt;- fit$sigmasq[1]
  }
  truth_sigma2_std &lt;- data$var_epsilon[1, 1] / (sd_y^2)
  
  rv_text &lt;- c(&quot;Residual Variance:&quot;,
             paste(&quot;Estimate =&quot;, round(sigma2_std, 3)),
             paste(&quot;Truth =&quot;, round(truth_sigma2_std, 3)))
usr &lt;- par(&quot;usr&quot;)  # get plot region (x1, x2, y1, y2)
leg_x &lt;- usr[2] - diff(usr[1:2]) * 0.05
leg_y &lt;- usr[3] + diff(usr[3:4]) * 0.2
legend(x = leg_x, y = leg_y,
       legend = rv_text,
       bty = &quot;o&quot;,
       cex = 0.8,
       box.lwd = 1,
       #adj = -1,
       bg = &quot;white&quot;)
  
  invisible(NULL)
}

reproduce_simulation_data &lt;- function(X, seed_val, ncausal = 10, h2 = 0.3, 
                                        ld_threshold = 0.25, max_attempts = 10000) {
  # This function sets the seed and then calls generate_sparse_eqtl_data
  # to produce the same simulation data instance reproducibly.
  data_instance &lt;- generate_sparse_eqtl_data(X, 
                                             K = ncausal, 
                                             h2 = h2, 
                                             seed = seed_val, 
                                             ld_threshold = ld_threshold, 
                                             max_attempts = max_attempts)
  return(data_instance)
}

generate_sparse_eqtl_data &lt;- function(X,
                                      K = 10,      # Number of effect SNPs
                                      h2 = 0.3,    # Variance explained
                                      seed = NULL,
                                      ld_threshold = 0.25,
                                      max_attempts = 100) {
  if (!is.null(seed)) set.seed(seed)

  n_samples &lt;- nrow(X)
  n_features &lt;- ncol(X)

  # Initialize beta (effect sizes) to zero
  beta &lt;- rep(0, n_features)

  # Select causal indices that meet the low-LD condition
  causal_indices &lt;- get_valid_causal(G = X, ncausal = K,
                                     ld_threshold = ld_threshold,
                                     max_attempts = max_attempts)

  # Assign effects ~ N(0, 0.6^2) to the causal SNPs
  beta[causal_indices] &lt;- rnorm(K, mean = 0, sd = 0.6)

  # Compute genetic effects
  genetic_effect &lt;- X %*% beta

  # Calculate variance components for the desired heritability
  var_g &lt;- var(genetic_effect)           # Variance due to genetic effects
  var_e &lt;- var_g * (1 - h2) / h2           # Residual variance

  # Generate phenotype: y ~ N(Xβ, σ²)
  y &lt;- genetic_effect + rnorm(n_samples, mean = 0, sd = sqrt(var_e))

  # Return the generated data and simulation parameters
  return(list(
    X = X,
    y = y,
    beta = beta,
    causal_indices = causal_indices,
    var_epsilon = var_e,
    h2_input = h2,
    h2_estimated = var_g / (var_g + var_e)
  ))
}

get_valid_causal &lt;- function(G, ncausal, ld_threshold = 0.25, max_attempts = 100) {
  # G: genotype matrix (n x p)
  # ncausal: number of causal variants to select
  # ld_threshold: maximum allowed pairwise r^2 between causal variants
  # max_attempts: maximum attempts to find a valid set
  snp_indices &lt;- seq_len(ncol(G))
  for (attempt in 1:max_attempts) {
    causal_indices &lt;- sort(sample(snp_indices, ncausal))
    # Compute correlation among the selected SNPs
    corr_mat &lt;- cor(G[, causal_indices])
    # Zero out diagonal and lower triangle to check only off-diagonals
    corr_mat[lower.tri(corr_mat, diag = TRUE)] &lt;- 0
    # If maximum r^2 is below threshold, return these indices
    if (max(abs(corr_mat)) &lt; ld_threshold) {
      return(causal_indices)
    }
  }
  stop(&quot;Could not find a set of causal variants with LD (r^2) below &quot;,
       ld_threshold, &quot; after &quot;, max_attempts, &quot; attempts.&quot;)
}</code></pre>
<pre class="r"><code># Reproduce the data
data &lt;- reproduce_simulation_data(X_subset, seed_val = 8)</code></pre>
<div id="iteration-1" class="section level2">
<h2>Iteration 1</h2>
<p>For the first iteration, all three methods pick up a <strong>single,
false positive set</strong>. The LD matrix below shows the variants
captured in CS as columns and the causal variants as rows. Every variant
in this set is in moderate LD with both one of the largest variants (ID
= 667, |r| <span class="math inline">\(\approx\)</span> 0.6) by
magnitude. Also, these variants are in partial LD (|r| <span
class="math inline">\(\approx\)</span> 0.4) with two other causal
variants (IDs = 326, 741).</p>
<pre class="r"><code># Only run SuSiE because all methods are equivalent for the first iteration
susie_fit &lt;- susie(X = data$X, 
      y = data$y,
      L = 20,
      standardize = T,
      intercept = T,
      max_iter = 1)

#susie_fit$sets$cs # Single CS

# LD with causal variants
LD_mat[data$causal_indices, susie_fit$sets$cs %&gt;% unlist()] %&gt;% round(.,2) %&gt;% abs()</code></pre>
<pre><code>     550  552  627  634  640  677  680  683
116 0.13 0.13 0.13 0.12 0.12 0.12 0.12 0.11
133 0.13 0.13 0.13 0.14 0.14 0.14 0.14 0.13
196 0.04 0.05 0.04 0.03 0.03 0.03 0.03 0.04
227 0.09 0.09 0.09 0.08 0.08 0.08 0.08 0.07
326 0.41 0.41 0.39 0.36 0.36 0.36 0.36 0.36
360 0.14 0.14 0.14 0.15 0.15 0.15 0.15 0.15
441 0.11 0.11 0.12 0.11 0.11 0.11 0.11 0.08
667 0.60 0.60 0.61 0.64 0.64 0.63 0.63 0.62
726 0.12 0.12 0.12 0.12 0.12 0.12 0.13 0.12
741 0.43 0.44 0.43 0.44 0.44 0.45 0.45 0.44</code></pre>
</div>
<div id="iteration-2" class="section level2">
<h2>Iteration 2</h2>
<p>For the second iteration, both SuSiE and SuSiE-ash have 3 total CS (2
true, 1 false positive). <strong>Note this false positive set is
different from the first iteration</strong>. The original false positive
from iteration 1 was shrunk. SuSiE-inf captures a single CS containing a
causal variant. All three methods overestimate the residual variance,
however, SuSiE-inf is closet to the truth, followed by SuSiE.ash, and
then SuSiE.</p>
<pre class="r"><code>susie_fit &lt;- susie(X = data$X, # 8 iterations to convergence
      y = data$y,
      L = 20,
      standardize = T,
      intercept = T,
      max_iter = 2)

#susie_fit$sets$cs # 3 CS

susie_inf_fit &lt;- susie_inf(X = scale(data$X), # 7 iterations to convergence
                           y = scale(data$y),
                           L = 20,
                           verbose = F,
                           coverage = 0.95,
                           XtX = XtX,
                           LD = LD,
                           V = V,
                           Dsq = Dsq,
                           maxiter = 2)

#susie_inf_fit$sets # 1 CS

susie_ash_fit &lt;- susie_ash_RE_Marg(X = scale(data$X), # 21 iterations to convergence
                           y = scale(data$y),
                           L = 20,
                           verbose = F,
                           coverage = 0.95,
                           update_ash_sigma = F,
                           K.length = 20,
                           upper_bound = 2,
                           XtX = XtX,
                           LD = LD,
                           V = V,
                           Dsq = Dsq,
                           VtXt = VtXt,
                           maxiter = 2)

#susie_ash_fit$sets # 3 CS

# For the original SuSiE model (with susie_fit$alpha as an L x p matrix):
plot_susie_fit(susie_fit, data, model = &quot;original&quot;)</code></pre>
<p><img src="figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexmccreight/SuSiE-ASH/blob/acb1163e2abae63ee7289861ab6d9e9854f592af/docs/figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-5-1.png" target="_blank">acb1163</a>
</td>
<td>
alexmccreight
</td>
<td>
2025-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># For the SuSiE-inf model (with susie_inf_fit$PIP as a p x L matrix):
plot_susie_fit(susie_inf_fit, data, model = &quot;inf&quot;)</code></pre>
<p><img src="figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-2">
Past versions of unnamed-chunk-5-2.png
</button>
</p>
<div id="fig-unnamed-chunk-5-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexmccreight/SuSiE-ASH/blob/acb1163e2abae63ee7289861ab6d9e9854f592af/docs/figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-5-2.png" target="_blank">acb1163</a>
</td>
<td>
alexmccreight
</td>
<td>
2025-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># For the SuSiE-ash model (with susie_ash_fit$PIP as a p x L matrix):
plot_susie_fit(susie_ash_fit, data, model = &quot;ash&quot;)</code></pre>
<p><img src="figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-3">
Past versions of unnamed-chunk-5-3.png
</button>
</p>
<div id="fig-unnamed-chunk-5-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexmccreight/SuSiE-ASH/blob/acb1163e2abae63ee7289861ab6d9e9854f592af/docs/figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-5-3.png" target="_blank">acb1163</a>
</td>
<td>
alexmccreight
</td>
<td>
2025-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="iteration-3" class="section level2">
<h2>Iteration 3</h2>
<p>For the third iteration, SuSiE and SuSiE.ash both introduce an
additional false positive CS of size 53 and 52, respectively while
retaining the 3 CS from the previous iteration. All of these variants in
the new CS are in moderately high LD with a causal variant (ID = 741,
|r| <span class="math inline">\(\approx\)</span> 0.65). Additionally,
these variants are all in moderate LD (|r| <span
class="math inline">\(\approx\)</span> 0.5) with the variants in the
first false positive CS. Finally, SuSiE-inf no longer has any credible
sets past this iteration.</p>
<pre class="r"><code>susie_fit &lt;- susie(X = data$X, # 8 iterations to convergence
      y = data$y,
      L = 20,
      standardize = T,
      intercept = T,
      max_iter = 3)

#susie_fit$sets$cs # 4 CS

susie_inf_fit &lt;- susie_inf(X = scale(data$X), # 7 iterations to convergence
                           y = scale(data$y),
                           L = 20,
                           verbose = F,
                           coverage = 0.95,
                           XtX = XtX,
                           LD = LD,
                           V = V,
                           Dsq = Dsq,
                           maxiter = 3)

#susie_inf_fit$sets # 0 CS

susie_ash_fit &lt;- susie_ash_RE_Marg(X = scale(data$X), # 21 iterations to convergence
                           y = scale(data$y),
                           L = 20,
                           verbose = F,
                           coverage = 0.95,
                           update_ash_sigma = F,
                           K.length = 20,
                           upper_bound = 2,
                           XtX = XtX,
                           LD = LD,
                           V = V,
                           Dsq = Dsq,
                           VtXt = VtXt,
                           maxiter = 3)

#susie_ash_fit$sets # 4 CS

# LD between variants in CS4 and CS1
#LD_mat[susie_fit$sets$cs$L4 %&gt;% unlist(), susie_fit$sets$cs$L1 %&gt;% unlist()] %&gt;% round(.,3) %&gt;% abs()

# LD between variants in CS4 and causal variants
#LD_mat[susie_fit$sets$cs$L4 %&gt;% unlist(), data$causal_indices] %&gt;% round(.,3) %&gt;% abs()

# For the original SuSiE model (with susie_fit$alpha as an L x p matrix):
plot_susie_fit(susie_fit, data, model = &quot;original&quot;)</code></pre>
<p><img src="figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexmccreight/SuSiE-ASH/blob/acb1163e2abae63ee7289861ab6d9e9854f592af/docs/figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-6-1.png" target="_blank">acb1163</a>
</td>
<td>
alexmccreight
</td>
<td>
2025-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># For the SuSiE-inf model (with susie_inf_fit$PIP as a p x L matrix):
plot_susie_fit(susie_inf_fit, data, model = &quot;inf&quot;)</code></pre>
<p><img src="figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-2">
Past versions of unnamed-chunk-6-2.png
</button>
</p>
<div id="fig-unnamed-chunk-6-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexmccreight/SuSiE-ASH/blob/acb1163e2abae63ee7289861ab6d9e9854f592af/docs/figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-6-2.png" target="_blank">acb1163</a>
</td>
<td>
alexmccreight
</td>
<td>
2025-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># For the SuSiE-ash model (with susie_ash_fit$PIP as a p x L matrix):
plot_susie_fit(susie_ash_fit, data, model = &quot;ash&quot;)</code></pre>
<p><img src="figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-6-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-3">
Past versions of unnamed-chunk-6-3.png
</button>
</p>
<div id="fig-unnamed-chunk-6-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexmccreight/SuSiE-ASH/blob/acb1163e2abae63ee7289861ab6d9e9854f592af/docs/figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-6-3.png" target="_blank">acb1163</a>
</td>
<td>
alexmccreight
</td>
<td>
2025-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="final-iteration" class="section level2">
<h2>Final Iteration</h2>
<p>SuSiE converged after 8 iterations, SuSiE-inf after 7, and SuSiE.ash
after 22. SuSiE and SuSiE.ash retained their 4 CS, although slightly
reduced the sizes of them. SuSiE-inf not only didn’t have a single CS,
it’s maximum PIP value (alpha value in SuSiE) was only 0.001. Thus, it
was so uncertain it shrunk every effect to null.</p>
<pre class="r"><code>susie_fit &lt;- susie(X = data$X, # 8 iterations to convergence
      y = data$y,
      L = 20,
      standardize = T,
      intercept = T,
      max_iter = 100)

#susie_fit$sets$cs # 4 CS

susie_inf_fit &lt;- susie_inf(X = scale(data$X), # 7 iterations to convergence
                           y = scale(data$y),
                           L = 20,
                           verbose = F,
                           coverage = 0.95,
                           XtX = XtX,
                           LD = LD,
                           V = V,
                           Dsq = Dsq,
                           maxiter = 100)

#susie_inf_fit$sets # 0 CS
# max(susie_inf_fit$PIP) 

susie_ash_fit &lt;- susie_ash_RE_Marg(X = scale(data$X), # 21 iterations to convergence
                           y = scale(data$y),
                           L = 20,
                           verbose = F,
                           coverage = 0.95,
                           update_ash_sigma = F,
                           K.length = 20,
                           upper_bound = 2,
                           XtX = XtX,
                           LD = LD,
                           V = V,
                           Dsq = Dsq,
                           VtXt = VtXt,
                           maxiter = 100)

#susie_ash_fit$sets # 4 CS

# For the original SuSiE model (with susie_fit$alpha as an L x p matrix):
plot_susie_fit(susie_fit, data, model = &quot;original&quot;)</code></pre>
<p><img src="figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexmccreight/SuSiE-ASH/blob/acb1163e2abae63ee7289861ab6d9e9854f592af/docs/figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-7-1.png" target="_blank">acb1163</a>
</td>
<td>
alexmccreight
</td>
<td>
2025-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># For the SuSiE-inf model (with susie_inf_fit$PIP as a p x L matrix):
#plot_susie_fit(susie_inf_fit, data, model = &quot;inf&quot;)

# For the SuSiE-ash model (with susie_ash_fit$PIP as a p x L matrix):
plot_susie_fit(susie_ash_fit, data, model = &quot;ash&quot;)</code></pre>
<p><img src="figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-2">
Past versions of unnamed-chunk-7-2.png
</button>
</p>
<div id="fig-unnamed-chunk-7-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/alexmccreight/SuSiE-ASH/blob/acb1163e2abae63ee7289861ab6d9e9854f592af/docs/figure/vignette_iteration_analysis_1.Rmd/unnamed-chunk-7-2.png" target="_blank">acb1163</a>
</td>
<td>
alexmccreight
</td>
<td>
2025-02-26
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31)
Platform: aarch64-apple-darwin20
Running under: macOS Sonoma 14.1

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib 
LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/New_York
tzcode source: internal

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] RColorBrewer_1.1-3  susieR_0.14.4       testthat_3.2.3     
 [4] mr.ash.alpha_0.1-43 RcppEigen_0.3.4.0.2 Rcpp_1.0.14        
 [7] lubridate_1.9.4     forcats_1.0.0       stringr_1.5.1      
[10] dplyr_1.1.4         purrr_1.0.4         readr_2.1.5        
[13] tidyr_1.3.1         tibble_3.2.1        ggplot2_3.5.1      
[16] tidyverse_2.0.0     workflowr_1.7.1    

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.1       fastmap_1.2.0          reshape_0.8.9         
 [4] promises_1.3.2         digest_0.6.37          timechange_0.3.0      
 [7] mime_0.12              lifecycle_1.0.4        ellipsis_0.3.2        
[10] processx_3.8.5         magrittr_2.0.3         compiler_4.4.2        
[13] rlang_1.1.5            sass_0.4.9             tools_4.4.2           
[16] yaml_2.3.10            knitr_1.49             htmlwidgets_1.6.4     
[19] pkgbuild_1.4.6         plyr_1.8.9             pkgload_1.4.0         
[22] miniUI_0.1.1.1         withr_3.0.2            desc_1.4.3            
[25] grid_4.4.2             urlchecker_1.0.1       git2r_0.35.0          
[28] profvis_0.4.0          xtable_1.8-4           colorspace_2.1-1      
[31] scales_1.3.0           cli_3.6.4              rmarkdown_2.29        
[34] crayon_1.5.3           generics_0.1.3         remotes_2.5.0         
[37] RcppParallel_5.1.10    rstudioapi_0.17.1      httr_1.4.7            
[40] tzdb_0.4.0             RcppArmadillo_14.2.3-1 sessioninfo_1.2.3     
[43] cachem_1.1.0           matrixStats_1.5.0      vctrs_0.6.5           
[46] devtools_2.4.5         Matrix_1.7-1           jsonlite_1.8.9        
[49] callr_3.7.6            hms_1.1.3              mixsqp_0.3-54         
[52] irlba_2.3.5.1          jquerylib_0.1.4        glue_1.8.0            
[55] ps_1.8.1               stringi_1.8.4          gtable_0.3.6          
[58] later_1.4.1            RcppZiggurat_0.1.6     munsell_0.5.1         
[61] pillar_1.10.1          brio_1.1.5             htmltools_0.5.8.1     
[64] R6_2.6.1               rprojroot_2.0.4        evaluate_1.0.3        
[67] shiny_1.10.0           lattice_0.22-6         Rfast_2.1.4           
[70] memoise_2.0.1          httpuv_1.6.15          bslib_0.9.0           
[73] whisker_0.4.1          xfun_0.50              fs_1.6.5              
[76] usethis_3.1.0          getPass_0.2-4          pkgconfig_2.0.3       </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
